tests/unit/s2n_resume_test.c

```diff
@@ -0,0 +1,253 @@
+        }
+
+        /* Erroneous secret size */
+        {
+            struct s2n_connection *conn;
+            EXPECT_NOT_NULL(conn = s2n_connection_new(S2N_SERVER));
+            conn->actual_protocol_version = S2N_TLS13;
+
+            DEFER_CLEANUP(struct s2n_stuffer output = { 0 }, s2n_stuffer_free);
+            EXPECT_SUCCESS(s2n_stuffer_growable_alloc(&output, 0));
+
+            struct s2n_ticket_fields ticket_fields = { .ticket_age_add = 1, .session_secret = test_session_secret };
+
+            ticket_fields.session_secret.size = UINT8_MAX + 1;
+            EXPECT_ERROR_WITH_ERRNO(s2n_tls13_serialize_resumption_state(conn, &ticket_fields, &output), S2N_ERR_SAFETY);
+
+            EXPECT_SUCCESS(s2n_connection_free(conn));
+        }
+    }
+
+    /* s2n_encrypt_session_ticket */
+    {
+        /* Session ticket keys. Taken from test vectors in https://tools.ietf.org/html/rfc5869 */
+        uint8_t ticket_key_name[16] = "2016.07.26.15\0";
+        S2N_BLOB_FROM_HEX(ticket_key, 
+        "077709362c2e32df0ddc3f0dc47bba6390b6c73bb50f9c3122ec844ad7c2b3e5");
+
+        /* Check encrypted data can be decrypted correctly for TLS12 */
+        {
+            struct s2n_connection *conn;
+            struct s2n_config *config;
+            uint64_t current_time;
+            EXPECT_NOT_NULL(conn = s2n_connection_new(S2N_SERVER));
+            EXPECT_NOT_NULL(config = s2n_config_new());
+
+            EXPECT_SUCCESS(s2n_config_set_session_tickets_onoff(config, 1));
+            EXPECT_SUCCESS(config->wall_clock(config->sys_clock_ctx, &current_time));
+            EXPECT_SUCCESS(s2n_config_add_ticket_crypto_key(config, ticket_key_name, strlen((char *)ticket_key_name),
+                         ticket_key.data, sizeof(ticket_key), current_time/ONE_SEC_IN_NANOS));
+
+            EXPECT_SUCCESS(s2n_connection_set_config(conn, config));
+            conn->actual_protocol_version = S2N_TLS12;
+
+            struct s2n_blob secret = { 0 };
+            struct s2n_stuffer secret_stuffer = { 0 };
+            EXPECT_SUCCESS(s2n_blob_init(&secret, conn->secure.master_secret, S2N_TLS_SECRET_LEN));
+            EXPECT_SUCCESS(s2n_stuffer_init(&secret_stuffer, &secret));
+            EXPECT_SUCCESS(s2n_stuffer_write_bytes(&secret_stuffer, test_master_secret.data, S2N_TLS_SECRET_LEN));
+            conn->secure.cipher_suite = &s2n_ecdhe_ecdsa_with_aes_128_gcm_sha256;
+
+            uint8_t data[S2N_TICKET_SIZE_IN_BYTES] = { 0 };
+            struct s2n_blob blob = { 0 };
+            struct s2n_stuffer output = { 0 };
+            EXPECT_SUCCESS(s2n_blob_init(&blob, data, sizeof(data)));
+            EXPECT_SUCCESS(s2n_stuffer_init(&output, &blob));
+
+            EXPECT_SUCCESS(s2n_encrypt_session_ticket(conn, NULL, &output));
+
+            /* Wiping the master secret to prove that the decryption function actually writes the master secret */
+            EXPECT_SUCCESS(s2n_stuffer_wipe(&secret_stuffer));
```

Comment:
This might be more readable / explicit if you operate on conn->secure.master_secret directly via memset. Otherwise, I have to prove to myself that wiping secret_stuffer does wipe the master secret on the connection.